protocols/api/src/main/java/org/apache/james/protocols/api/Encryption.java

```diff
@@ -29,15 +29,17 @@
     private final SSLContext context;
     private final boolean starttls;
     private final String[] enabledCipherSuites;
+    private Boolean clientAuth;
 
-    private Encryption(SSLContext context, boolean starttls, String[] enabledCipherSuites) {
+    private Encryption(SSLContext context, boolean starttls, String[] enabledCipherSuites, Boolean clientAuth) {
```

Comment:
Maybe an enum would be better than a Boolean.

Especially an enum would enable to explicit the usage of `null` that is unclear to me.

Reply:
Agreed, enums kind of are the Java way to express tri-state or multi-state logic. But as discussed below, I will remove the unused optional authentication feature for now, so this will become a simple `boolean`.

Reply:
Even with too states enums brings in strong typing thus better express the intent and should rather be used.

(Refs: Clean code book, chapter I don't remember but clearly boolean arguments should be discouraged.)

Reply:
Fixed in latest push.